lte/gateway/c/core/oai/tasks/amf/amf_authentication.cpp

```diff
@@ -0,0 +1,1073 @@
+ ** Inputs:  ue_id:      UE lower layer identifier                         **
+ **      emm_cause: Authentication failure AMF cause code                  **
+ **      res:       Authentication response parameter. or auts             **
+ **                 in case of sync failure                                **
+ **      Others:    None                                                   **
+ **                                                                        **
+ ** Outputs:     None                                                      **
+ **      Return:    RETURNok, RETURNerror                                  **
+ **      Others:    amf_data, T3560                                        **
+ **                                                                        **
+ ***************************************************************************/
+int amf_proc_authentication_complete(
+    amf_ue_ngap_id_t ue_id, AuthenticationResponseMsg* msg, int amf_cause,
+    const unsigned char* res) {
+  OAILOG_FUNC_IN(LOG_NAS_AMF);
+  int rc                         = RETURNerror;
+  bool is_xres_validation_failed = false;
+  nas_amf_smc_proc_t nas_amf_smc_proc_autn;
+  nas_amf_registration_proc_t* registration_proc = NULL;
+  nas5g_amf_auth_proc_t* auth_proc               = NULL;
+
+  OAILOG_DEBUG(
+      LOG_NAS_AMF,
+      "Authentication  procedures complete for "
+      "(ue_id=" AMF_UE_NGAP_ID_FMT ")\n",
+      ue_id);
+  ue_m5gmm_context_s* ue_mm_context = NULL;
+
+  amf_context_t* amf_ctx = NULL;
+  ue_mm_context          = amf_ue_context_exists_amf_ue_ngap_id(ue_id);
+
+  if (!ue_mm_context) {
+    OAILOG_WARNING(
+        LOG_NAS_AMF,
+        "AMF-PROC - Failed to authenticate the UE due to NULL"
+        "ue_mm_context\n");
+    OAILOG_FUNC_RETURN(LOG_NAS_AMF, rc);
+  }
+
+  amf_ctx = &ue_mm_context->amf_context;
+
+  registration_proc = get_nas_specific_procedure_registration(amf_ctx);
+  auth_proc         = get_nas5g_common_procedure_authentication(amf_ctx);
+
+  if (auth_proc) {
+    /*    Stop Timer T3560 */
+    OAILOG_INFO(
+        LOG_NAS_EMM,
+        "Timer:  Stopping Authentication Timer T3560 with id = %lu\n",
+        auth_proc->T3560.id);
+    amf_app_stop_timer(auth_proc->T3560.id);
+    auth_proc->T3560.id = NAS5G_TIMER_INACTIVE_ID;
+    OAILOG_INFO(LOG_NAS_AMF, "Timer: After Stopping T3560 Timer\n");
+
+    nas_amf_smc_proc_autn.amf_ctx_set_security_eksi(amf_ctx, auth_proc->ksi);
+    registration_proc->ksi = auth_proc->ksi;
+
+#if 0  // commented as part of warning cleanup, need to identify correct way to
+       // print
+    OAILOG_STREAM_HEX(
+        OAILOG_LEVEL_TRACE, LOG_AMF_APP, "Received RES*: ",
+        (const char*) (&(msg->autn_response_parameter.response_parameter[0])),
+        AUTH_XRES_SIZE);
+
+    OAILOG_STREAM_HEX(
+        OAILOG_LEVEL_TRACE, LOG_AMF_APP, "Expected RES*: ",
+        (const char*) (&(amf_ctx->_vector[auth_proc->ksi].xres_star[0])),
+        AUTH_XRES_SIZE);
+#endif
```

Comment:
remove if unused. Instead put a TODO comment about showing debug trace.

Reply:
comment addressed.
removed, will add debug traces in log cleanup activity(#8266)